pkg/kube/pod_controller.go

```diff
@@ -99,11 +99,17 @@ func NewPodController(cli kubernetes.Interface, options *PodOptions, opts ...Pod
 	return r
 }
 
-// NewPodControllerForExistingPod returns a new PodController given Kubernetes
-// Client and existing pod details.
-// Invocation of StartPod of returned PodController instance will fail, since the pod is already known.
-func NewPodControllerForExistingPod(cli kubernetes.Interface, pod *corev1.Pod) PodController {
-	r := &podController{
+// NewPodControllerForExistingPod returns a new PodController for the given
+// running pod.
+// Invocation of StartPod of returned PodController instance will fail, since
+// the pod is expected to be running already.
```

Comment:
```suggestion
// the pod is expected to be running already.

// Note:
// If the pod is not in the ready state, it will wait for up to
// KANISTER_POD_READY_WAIT_TIMEOUT (15 minutes by default) until the pod becomes ready.
```